### PR TITLE
Added rule about property observers (willSet and didSet)

### PR DESCRIPTION
**Pull request by ThibaultKlein**

### CODE REVIEW

**Comment by ThibaultKlein** [comment]
@paulmiard @Haud

**Comment by Haud** [comment]
This is good and I don't disagree with anything here; however, as is, I am not sure this is particularly useful for a style guide. As of now, this doesn't give any justification of why to use it or even give a didactic `must use` statement, and instead just describes what it is and how to use -- which again, isn't _bad_, but I don't see the utility of this for now. 

Are there places where we can delineate must-use cases? If not, I vote we jettison making this a proper ruleset and just educate on its utility. Maybe we can have a `Tips & Tricks` subsection that this can go in because I think this might make more sense there. Thoughts @paulmiard ?

**Comment by ThibaultKlein** [comment]
That's cool with me to have a tips section, I don't know how useful property observers can be for the style guide since it's not really a coding style ;).

**Comment by Haud** [comment]
Yea that's my point -- this is nice and useful information but its' not style-guide information, so I don't think it belongs here.

**Comment by paulmiard** [comment]
Yeah that works. I'd rather call it `Best Practices` though. Sounds better than `Tips & Tricks`. Would that work? Might even be worth moving it to a different file. But we can do later as well

**Comment by ThibaultKlein** [comment]
I updated the file to include a `Best Practices` section and have the property observers inside (it's still in the same file for now)

**Comment by Haud** [comment]
I think we should just move it to its own file. We can discuss it today.

File: README.md [hunk 1]
```diff
@@ -133,3 +133,55 @@ var intValue: Int
...
+```
+
+You can use 2 types of property observers:
+
+* *willSet* is called just before the property value is stored.
+* *didSet* is called right after the property value is stored.
+
+```swift
+var intValue: Int = 0 {
+	willSet(newIntValue) {
```
Comment (Haud) [inline README.md]: Don't need to use `newIntValue`. It should just be `willSet` and use `newValue`

File: README.md [hunk 2]
```diff
@@ -133,3 +133,55 @@ var intValue: Int
...
+		print("About to set intValue to \(newIntValue)")
+  	}
+    didSet {
```
Comment (Haud) [inline README.md]: `oldValue`

**Comment by paulmiard** [comment]
rebase? move in its own file? both?

**Comment by ThibaultKlein** [comment]
Rebased, created the separate file, added a link to it in the README

**Comment by Haud** [comment]
Looks good :+1:

**Comment by ThibaultKlein** [comment]
@paulmiard any comments or can we merge?

**Comment by paulmiard** [comment]
:+1:

**Comment by paulmiard** [comment]
I like approving PRs after they get merged

**Pull request merged**